test/e2e/node_reuse_test.go

```diff
@@ -509,15 +511,37 @@ func untaintNodes(clientSet *kubernetes.Clientset, nodes *corev1.NodeList, taint
 		Logf("Untainting node %v ...", nodes.Items[i].Name)
 		newNode, changed := removeTaint(&nodes.Items[i], taint)
 		if changed {
-			node, err := clientSet.CoreV1().Nodes().Update(ctx, newNode, metav1.UpdateOptions{})
-			Expect(err).To(BeNil(), "Failed to update nodes")
-			Logf("Node %v untainted", node.Name)
+			err := patchNode(clientSet, nodes.Items[i].Name, &nodes.Items[i], newNode)
```

Comment:
See my other comment for more explanations.
```suggestion
                        patchBytes := generatePatch(oldNode *corev1.Node, newNode *corev1.Node)
			err := clientSet.CoreV1().Nodes().Patch(ctx, nodeName, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
```